radix-transactions/src/manifest/static_resource_movements/typed_invocation.rs

```diff
@@ -1,622 +1,1127 @@
                     ),*
                 }
 
-                $(
-                    // For each blueprint we define a type that's made up of the method and function
-                    #[derive(Debug)]
-                    pub enum [< $blueprint_name BlueprintInvocations >] {
-                        Function([< $blueprint_name Function >]),
-                        Method([< $blueprint_name Method >]),
-                    }
-
-                    #[derive(Debug)]
-                    pub enum [< $blueprint_name Method >] {
-                        $(
-                            $method_ident($method_input)
-                        ),*
-                    }
-
-                    impl [< $blueprint_name Method >] {
-                        #[allow(unreachable_patterns)]
-                        pub fn method_name(&self) -> &str {
-                            match self {
-                                $(
-                                    Self::$method_ident(..) => $method_name,
-                                )*
-                                // AVOIDS [E0004] when the enum is empty "note: references are always considered inhabited"
-                                // https://github.com/rust-lang/unsafe-code-guidelines/issues/413
-                                _ => unreachable!()
-                            }
-                        }
-
-                        pub fn from_invocation(
-                            method_name: &str,
-                            args: &::radix_common::prelude::ManifestValue
-                        ) -> Result<Self, StaticResourceMovementsError> {
-                            Ok(match method_name {
-                                $(
-                                    $method_name => {
-                                        let encoded = ::radix_common::prelude::manifest_encode(args)
-                                            .map_err(StaticResourceMovementsError::NativeArgumentsEncodeError)?;
-                                        let decoded = ::radix_common::prelude::manifest_decode(&encoded)
-                                            .map_err(StaticResourceMovementsError::NativeArgumentsDecodeError)?;
-                                        Self::$method_ident(decoded)
-                                    }
-                                )*
-                                _ => return Err(StaticResourceMovementsError::UnknownNativeMethod {
-                                    package: $package_address,
-                                    blueprint: stringify!($blueprint_name).to_string(),
-                                    method: method_name.to_string(),
-                                })
+                impl [< $blueprint_ident BlueprintMethod >] {
+                    #![allow(unreachable_patterns, unused_variables)]
+                    pub fn decode_invocation(
+                        // Only provided for nice errors. Not needed in the decoding.
```

Comment:
SUPER MINOR: I don't think we really need it. The `blueprint_id` will capture e.g. "RoyaltyModule" or something in the case of a non-main blueprint.

Reply:
That's fair, I will remove the module id here.